src/facade/cmd_arg_parser.h

```diff
@@ -123,10 +123,19 @@ struct CmdArgParser {
   }
 
   // Consume next value
-  NextProxy Next() {
+  template <class T = NextProxy> auto Next() {
     if (cur_i_ >= args_.size())
       Report(OUT_OF_BOUNDS, cur_i_);
-    return NextProxy{this, cur_i_++};
+
+    NextProxy next{this, cur_i_++};
+    if constexpr (std::is_arithmetic_v<T>) {
+      return next.Num<T>();
+    } else if constexpr (std::is_convertible_v<std::string_view, T>) {
+      return static_cast<T>(next);
+    } else {
+      static_assert(std::is_same_v<T, NextProxy>, "incorrect type");
+      return next;
+    }
   }
```

Comment:
1. Let's slice off the last commits and move them to a new PR
2. Now NextProxy's task is only to convert to strings and provide a binding to the caseproxy with is a little weird 🙂

Reply:
As we discussed I will do some more fixes in separate PR